### DISCUSSION TITLE
TraceQL array not able to find specific item

### DISCUSSION THREAD

**Discussion by hazra-mms** [grafana/tempo#3773] [Q&A] [answered]
Hi,

We are using Tempo 2.5 in our cluster, we have http headers which are attached as span attribute. Those are marked as array and we are trying to use the new Tempo 2.5 array query for searching flow id. But no matter what, we always get an empty response.

The query we use in TraceQL is `{span.http.request.header.x-flow-id[]=<uuid_value>}`

![image](https://github.com/grafana/tempo/assets/102611635/9e28e1fa-082a-48dd-a27c-5ebd70c609d6)

Could someone help to identify if there any issue with the query syntax?

The span attribute as viewed in tempo
![image](https://github.com/grafana/tempo/assets/102611635/cfaeb9d8-d0a7-4d5f-9ab5-b0f46cf1baef)

**Comment by joe-elliott** [comment, marked as answer]
Only vParquet4 (and future) blocks can search inside arrays. Can you try changing your block format to vParquet4?

https://grafana.com/docs/tempo/latest/configuration/parquet/

Something like this:
```
storage:
    trace:
        wal:
            version: vParquet4
        block:
            version: vParquet4
```

If you do adjust your block format I'd recommend removing this config option in future versions so you use the default block format. Otherwise you'll pin yourself to an out of date format.

**Reply by hazra-mms** [reply]
Thank you, works perfectly 👍
